product_cost_security/tests/test_product_cost_security.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2018-2020 ForgeFlow, S.L.
+# Copyright 2018-2019 Brainbean Apps (https://brainbeanapps.com)
+# Copyright 2018-2019 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import Form, TransactionCase
+
+
+class TestProductCostSecurity(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product_edit_cost_group_id = cls.env.ref(
+            "product_cost_security.group_product_edit_cost"
+        ).id
+        cls.product_cost_group_id = cls.env.ref(
+            "product_cost_security.group_product_cost"
+        ).id
+        cls.product_template_model = cls.env["product.template"].with_context(
+            tracking_disable=True
+        )
+        cls.env.user.write(
+            {
+                "groups_id": [
+                    (6, 0, cls.env.ref("base.group_system").ids),
+                ],
+            }
+        )
+
+    def test_0(self):
+        sheet_form = Form(self.product_template_model.with_user(self.env.user))
+        sheet_form.name = "Test Product"
+        sheet_form.save()
+        with self.assertRaises(AssertionError):
+            # It would raise an AssertionError as the field would not be found in the view
+            # as the user does not have the group- Access to product costs
+            standard_price = sheet_form.standard_price
+        self.env.user.groups_id = [
+            (
+                6,
+                0,
+                [self.product_cost_group_id],
+            )
+        ]
+        sheet_form = Form(self.product_template_model.with_user(self.env.user))
+        sheet_form.name = "Test Product"
+        sheet_form.save()
+        standard_price = (
+            sheet_form.standard_price
+        )  # It would not raise any error now as the user
+        # has the required group to view the costs
+        self.assertEqual(standard_price, 0.0)
+        with self.assertRaises(AssertionError):
+            # It would raise an AssertionError as the field is readonlyin the view
+            # as the user does not have the group- Modify product costs
+            sheet_form.standard_price = 5.0
+        self.env.user.groups_id = [
+            (
+                6,
+                0,
+                [self.product_edit_cost_group_id],
+            )
+        ]
+        sheet_form = Form(self.product_template_model.with_user(self.env.user))
```

Comment:
You are testing three different cases here, so you could split this test and have 3 tests in total

Reply:
Hello @astirpe , I have made the changes as per your suggestions.There are now 7 tests in all, as I had to add a field 'user_can_see_standard_price_uom' for product.product (product variant) as well.

Thanks